velox/connectors/hive/tests/PartitionIdGeneratorTest.cpp

```diff
@@ -189,4 +189,60 @@ TEST_F(PartitionIdGeneratorTest, limitOfPartitionNumber) {
       fmt::format("Exceeded limit of {} distinct partitions.", maxPartitions));
 }
 
+TEST_F(PartitionIdGeneratorTest, supportedPartitionKeyTypes) {
+  // Test on supported key types.
+  {
+    PartitionIdGenerator idGenerator(
+        ROW({
+            VARCHAR(),
+            BOOLEAN(),
+            VARBINARY(),
+            TINYINT(),
+            SMALLINT(),
+            INTEGER(),
+            BIGINT(),
+        }),
+        {0, 1, 2, 3, 4, 5, 6},
+        100,
+        pool());
+
+    auto input = makeRowVector({
+        vectorMaker_.flatVectorNullable<StringView>(
+            {"Left", std::nullopt, "Right"}, VARCHAR()),
+        vectorMaker_.flatVectorNullable<bool>({true, false, std::nullopt}),
+        vectorMaker_.flatVector<StringView>(
+            {"proton", "neutron", "electron"}, VARBINARY()),
+        vectorMaker_.flatVectorNullable<int8_t>({1, 2, std::nullopt}),
+        vectorMaker_.flatVectorNullable<int16_t>({1, 2, std::nullopt}),
+        vectorMaker_.flatVectorNullable<int32_t>({1, std::nullopt, 2}),
+        vectorMaker_.flatVectorNullable<int64_t>({std::nullopt, 1, 2}),
+    });
+
+    raw_vector<uint64_t> ids;
+    idGenerator.run(input, ids);
+
+    EXPECT_TRUE(ids[0] == 0 && ids[1] == 1 && ids[2] == 2);
+  }
+
+  // Test unsupported partition key types.
+  {
+    auto input = makeRowVector({
+        vectorMaker_.constantVector<float>({1.0}),
+        vectorMaker_.constantVector<double>({1.0}),
+        vectorMaker_.constantVector<Timestamp>(
+            {Timestamp::fromMillis(1639426440000)}),
+        vectorMaker_.arrayVector<int32_t>({{1, 2, 3}}),
+        vectorMaker_.mapVector<int16_t, int16_t>({{{1, 2}}}),
+    });
+
+    for (column_index_t i = 1; i < input->childrenSize(); i++) {
```

Comment:
column_index_t -> auto

Reply:
Got complaint about casting int to unsigned int from the PartitionIdGenerator constructor at the next line, when changing it to auto. So decide to keep column_index_t.